src/VitessceGrid.js

```diff
@@ -1,137 +1,88 @@
-import React from 'react';
-
+import React, { useState } from 'react';
 import { Responsive, WidthProvider } from 'react-grid-layout';
-
 import { getMaxRows, resolveLayout } from './layoutUtils';
 
-function shallowEqual(objA, objB) {
-  // Taken from "shallowequal" on NPM, with modifications.
-  if (objA === objB) {
-    return true;
-  }
-  if (typeof objA !== 'object' || !objA || typeof objB !== 'object' || !objB) {
-    return false;
-  }
-
-  const keysA = Object.keys(objA);
-  const keysB = Object.keys(objB);
-  if (keysA.length !== keysB.length) {
-    return false;
-  }
-
-  for (let idx = 0; idx < keysA.length; idx++) { // eslint-disable-line no-plusplus
-    const key = keysA[idx];
-    if (!Object.prototype.hasOwnProperty.call(objB, key)) {
-      return false;
-    }
-    if (key === 'onAllReady') {
-      return true;
-      // TODO: We were stuck in an infinite loop because the fuctions were not equal.
-      // This is a hack.
-    }
-
-    const valueA = objA[key];
-    const valueB = objB[key];
-
-    if (valueA !== valueB) {
-      return false;
-    }
-  }
-
-  return true;
-}
-
-
-export default class VitessceGrid extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { readyComponentKeys: new Set() }; // eslint-disable-line react/no-unused-state
-    // This is being used and I'm not sure why there's a linting problem.
-  }
-
-  shouldComponentUpdate(nextProps) {
-    // We are ignoring state right now and only considering props.
-    // State is updated as components trigger onReady...
-    // but those events should not cause a re-render.
-    // React docs warn:
-    // > Do not rely on it to “prevent” a rendering, as this can lead to bugs.
-    // so there is probably a better approach.
-    return !shallowEqual(this.props, nextProps);
-  }
-
-  render() {
-    const {
-      layout, getComponent, padding, margin, draggableHandle,
-      reactGridLayoutProps, onAllReady, rowHeight,
-    } = this.props;
-
-    const ResponsiveGridLayout = WidthProvider(Responsive);
-
-    const {
-      cols, layouts, breakpoints, components,
-    } = resolveLayout(layout);
-
-    // Inline CSS is generally avoided, but this saves the end-user a little work,
-    // and prevents class names from getting out of sync.
-    const style = (
-      <style>
-        {`
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+export default function VitessceGrid(props) {
+  const {
+    layout, getComponent, padding, margin, draggableHandle,
+    reactGridLayoutProps, onAllReady, rowHeight,
+  } = props;
+  const {
+    cols, layouts, breakpoints, components,
+  } = resolveLayout(layout);
+
+  const readyComponentKeys = new Set(); // only created when props change due to React.Memo
```

Comment:
I changed this to a mutable var since I was having repeat renders with `useState`. I think your solution has fixed the re-rendering issue, and ideally we would want React to handle `readyComponentKeys` as well.

Since we removed `React.memo`, we don't want to initialize an empty set anytime the component re-renders. Would something like this work now, with your fix:

```javascript
const [readyComponentKeys, setReadyComponentKeys] = useState(new Set());
...
...
  const layoutChildren = Object.entries(gridComponents).map(([k, v]) => {
    const Component = getComponent(v.component);
    const onReady = setReadyComponetKeys((prevReadyComponentKeys) => {
      const nextReadyComponentKeys = new Set(prevReadyComponentKeys);
      nextReadyComponentKeys.add(k);
      if (nextReadyComponentKeys === Object.keys(gridComponents).length) {
        // The sets are now equal
        onAllReady();
      }
      return nextReadyComponentKeys;
    };

    const removeGridComponent = () => {
      // delete gridComponents[k];
      // setGridComponents(gridComponents);
      const newGridComponents = { ...gridComponents };
      delete newGridComponents[k];
      setGridComponents(newGridComponents);
    };
```

Reply:
I think I get it, but this doesn't seem quite right:
```
setReadyComponetKeys((prevReadyComponentKeys) => {
```
That would be passing an anonymous function as the argument?

I'll patch, and we can look at it in the morning.

Reply:
Never mind: Just read https://reactjs.org/docs/hooks-reference.html#functional-updates